HunterPie.Core/Game/World/Entities/Party/MHWPartyMember.cs

```diff
@@ -51,14 +53,31 @@ private set
         public event EventHandler<IPartyMember> OnDamageDealt;
         public event EventHandler<IPartyMember> OnWeaponChange;
 
+        /// <inheritdoc />
         void IUpdatable<MHWPartyMemberData>.Update(MHWPartyMemberData data)
         {
             Name = data.Name;
-            Damage = data.Damage;
+            if (data.Damage != 0)
+            {
+                _anyNonTrivialStatisticalDamage = true;
+                Damage = data.Damage;
+            }
```

Comment:
I think in this case the `if` condition is redundant

```suggestion
Name = data.Name;
_anyNonTrivialStatisticalDamage = data.Damage != 0;
Damage = data.Damage;
```

Reply:
If there is no damage data from party info, there will still be consecutive calls to this `IUpdatable<MHWPartyMemberData>.Update` function with `Damage == 0`. This will cause the accumulated damage number gather from native module getting cleared.